File: exercise/java/day08/src/test/java/ToyProductionServiceTest.java
import domain.Toy;
import doubles.InMemoryToyRepository;
import org.junit.jupiter.api.Test;
import service.ToyProductionService;

import static domain.Toy.State.*;
import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertEquals;


class ToyProductionServiceTest {
    private static final String TOY_NAME = "Train";

    @Test
    void assignToyToElfShouldPassTheItemInProduction() {
        var repository = new InMemoryToyRepository();
        var service = new ToyProductionService(repository);
        repository.save(new Toy(TOY_NAME, UNASSIGNED));

        service.assignToyToElf(TOY_NAME);

        assertThat(repository.findByName(TOY_NAME).getState())
                .isEqualTo(IN_PRODUCTION);
    }
}

File: exercise/kotlin/day08/src/test/kotlin/ToyProductionServiceTest.kt
package toyProduction

import domain.Toy
import domain.Toy.State.UNASSIGNED
import domain.Toy.State.IN_PRODUCTION
import doubles.InMemoryToyRepository
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import service.ToyProductionService

class ToyProductionServiceTest : StringSpec({
    val toyName = "Train"

    "assignToyToElfShouldPassTheItemInProduction" {
        val repository = InMemoryToyRepository()
        val service = ToyProductionService(repository)
        repository.save(Toy(toyName, UNASSIGNED))

        service.assignToyToElf(toyName)

        repository.findByName(toyName)?.state shouldBe IN_PRODUCTION
    }
})
